.circleci/config.yml

```diff
@@ -105,142 +280,104 @@ jobs:
       # workspaces, but that seemed to have a negligible effect on speed. It's
       # also tricky to pick specific sub directories since the API deploy
       # touches several top level folders {common,}api.
-      - restore_cache:
-          keys:
-          - api-cache-{{ checksum "~/workbench/api/build.gradle" }}
-          - api-cache-
-      - run: *activate_service_accont_creds
+      - api_cache:
+          restore: true
+      - activate_service_account_creds
       - deploy:
           name: Deploy to App Engine
           working_directory: ~/workbench/api
           command: |
             ./project.rb deploy \
               --project all-of-us-workbench-test \
-              --account circle-deploy-account@all-of-us-workbench-test.iam.gserviceaccount.com \
+              --account << pipeline.parameters.circleci_service_account >> \
               --version circle-ci-test \
               --key-file circle-sa-key.json \
               --promote
 
+
   api-deps-check:
     <<: *java_defaults
     steps:
-      - checkout
-      - run: *update_git_submodules
+      - git_checkout
       - run:
           name: Scan dependencies for vulnerabilities
           working_directory: ~/workbench/api
-          command: |
-            ./project.rb gradle dependencyCheckAnalyze --info
+          command: ./project.rb gradle dependencyCheckAnalyze --info
 
   api-integration-test:
     <<: *java_defaults
     steps:
-      - checkout
-      - run: *update_git_submodules
-      - run: *ensure_branch_has_api_changes
-      - restore_cache:
-          keys:
-          - api-integration-cache-{{ checksum "~/workbench/api/build.gradle" }}
-          - api-integration-cache-
-      - run: *activate_service_accont_creds
+      - git_checkout
+      - ensure_branch_has_changes
+      - api_cache:
+          restore: true
+      - activate_service_account_creds
       - run:
           name: Run Integration Tests
           working_directory: ~/workbench/api
           command: ./project.rb integration
       - store_test_results:
           path: ~/workbench/api/build/test-results/integrationTest/
-      - save_cache:
-          paths:
-            - ~/.gradle
-            - ~/.m2
-          key: api-integration-cache-{{ checksum "~/workbench/api/build.gradle" }}
+      - api_cache:
+          save: true
 
   api-nightly-integration-test:
     <<: *java_defaults
     steps:
-      - checkout
-      - run: *update_git_submodules
-      - restore_cache:
-          keys:
-          - api-integration-cache-{{ checksum "~/workbench/api/build.gradle" }}
-          - api-integration-cache-
-      - run: *activate_service_accont_creds
+      - git_checkout
+      - api_cache:
+          restore: true
+      - activate_service_account_creds
       - run:
           name: Run Nightly Integration tests
           working_directory: ~/workbench/api
           command: ./project.rb nightly-integration
```

Comment:
So is the "nightly" test suite a different one than the regular one? Or is this purely a scheduling thing?

If it's just for scheduling, I bet we don't need separate jobs, but just different triggers and maybe parameters.